tests/nested/manual/tpm/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Check that tpm works properly on UC20
+
+description: |
+  This test check UC20 can boot with secure boot successfully and UC18 doesn't boot with secure boot
+
+systems: [ubuntu-18.04-*, ubuntu-20.04-*]
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    create_nested_core_vm true
+    start_nested_core_vm 
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    cleanup_nested_env
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # Check tpm status on the nested vm
+    if is_core_20_nested_system; then
+        #TODO: check secure boot is enabled using test-snapd-mokutil snap
+        execute_remote "dmesg | grep -i tpm" | MATCH "efi: +SMBIOS=.* +TPMFinalLog=.*"
+        execute_remote "test -e /sys/kernel/security/tpm0/binary_bios_measurements"
+    else
+        execute_remote "dmesg | grep -i tpm" | MATCH "No TPM chip found"
+        execute_remote "! test -e /sys/kernel/security/tpm0/binary_bios_measurements"
+    fi
+
+    # Check the nested image runs the current branch  
+    execute_remote "systemctl is-active snapd.spread-tests-run-mode-tweaks" | MATCH "active"
```

Comment:
I think the test looks fine (but someone who knows about tpm should ack the details), my only wondering is about is_core_20_nested_system helper, per earlier comment.

Reply:
I'll remove this part of the code which is not needed anymore.